claimstore/modules/claims/restful.py

```diff
@@ -21,149 +21,306 @@
+        """Record a new claim.
 
-@claims_restful.route('/claims', methods=['POST'])
-@only_json
-def submit_claim():
-    """Record a new claim.
+        This POST service expects JSON data following the JSON schema defined
+        for claims.
+        """
+        json_data = request.get_json()
 
-    This POST service expects JSON data following the JSON schema defined for
-    claims.
-    """
-    json_data = request.get_json()
-
-    try:
-        validate_json(json_data, 'claims.claim')
-    except Exception as e:
-        raise InvalidUsage('JSON data is not valid', details=str(e))
-
-    # Check if claim.datetime is ISO 8601
-    try:
-        created_dt = isodate.parse_datetime(json_data['claim']['datetime'])
-    except isodate.ISO8601Error as e:
-        raise InvalidUsage(
-            'Claim datetime does not follow ISO 8601 Z',
-            details=str(e)
+        self.validate_json(json_data)
+
+        try:
+            created_dt = isodate.parse_datetime(json_data['claim']['created'])
+        except isodate.ISO8601Error as e:
+            raise InvalidJSONData(
+                'Claim `created` datetime does not follow ISO 8601 Z',
+                details=str(e)
+            )
+
+        claimant = Claimant.query.filter_by(name=json_data['claimant']).first()
+        if not claimant:
+            raise InvalidRequest('Claimant not registered')
+
+        subject_type = IdentifierType.query.filter_by(
+            name=json_data['subject']['type']
+        ).first()
+        if not subject_type:
+            raise InvalidRequest('Subject Type not registered')
+
+        object_type = IdentifierType.query.filter_by(
+            name=json_data['object']['type']
+        ).first()
+        if not object_type:
+            raise InvalidRequest('Object Type not registered')
+
+        if subject_type.id == object_type.id:
+            raise InvalidRequest('Subject and Object cannot have the same \
+                identifier type')
+
+        predicate = Predicate.query.filter_by(
+            name=json_data['claim']['predicate']
+        ).first()
+        if not predicate:
+            raise InvalidRequest('Predicate not registered')
+
+        arguments = json_data['claim'].get('arguments', {})
+        new_claim = Claim(
+            created=created_dt,
+            claimant=claimant,
+            subject_type_id=subject_type.id,
+            subject_value=json_data['subject']['value'],
+            predicate_id=predicate.id,
+            object_type_id=object_type.id,
+            object_value=json_data['object']['value'],
+            certainty=json_data['claim']['certainty'],
+            human=arguments.get('human', None),
+            actor=arguments.get('actor', None),
+            role=arguments.get('role', None),
+            claim_details=json_data,
         )
+        db.session.add(new_claim)
+        db.session.commit()
+        return {'status': 'success', 'uuid': new_claim.uuid}
+
+    def get(self):
```

Comment:
I guess it is because the DB is not being created by default. I have added the argument `--initdb` to `python run` and that should fix it.